### PR TITLE
[CR] A proof of concept via science

### PR DESCRIPTION
**Pull request by bkentel**
Ignoring the stuff between TODO and `mattack::science` (from line 491 down), how do you guys feel about this kind of style for functions? Namely, pulling the inline, unnamed constants out and to the top of the function to hopefully make tweaking balance easier; or to ease even moving the constants out into a .json file?

Pulling the strings out and to the top is maybe the most debatable point from my perspective. While it does consolidate them in one place, it separates them from the actual call site maybe too much?
- Added a function to return empty tiles within a given radius of a point or creature. No memory is allocated. (Todo: relocate)
- Added get_random_index to wrap rng when working with size_t values (localizes all the ugly casting). (Todo: relocate)
- The code is vertically longer, but hopefully clearer and more understandable at a glance.

### CODE REVIEW

File: src/monattack.cpp [hunk 1]
```diff
@@ -425,109 +425,258 @@ void mattack::smash(monster *z, int index)
...
+    // attack movement costs
+    constexpr int att_cost_shock   = 0;
+    constexpr int att_cost_rad     = 400;
+    constexpr int att_cost_manhack = 200;
+    constexpr int att_cost_acid    = 100;
+    constexpr int att_cost_flavor  = 80;
+    
+    // radiation attack behaviour
+    constexpr int att_rad_dodge_diff    = 16; // how hard it is to dodge
+    constexpr int att_rad_mutate_chance = 6;  // inverse chance to cause mutation.
```
Comment (narc0tiq) [inline src/monattack.cpp]: I think I'd like this comment to be more explicit, i.e. `// inverse chance (one in X) to cause mutation`. While the phrasing immediately clicked for me once I read it, I almost skimmed past it, and this doesn't behave like a direct chance (i.e. higher being more likely).

**Comment by narc0tiq** [comment]
I'm definitely on the side of keeping the strings where they're used. Having the tweakables at the top is nice, though.

File: src/monattack.cpp [hunk 2]
```diff
@@ -425,109 +425,258 @@ void mattack::smash(monster *z, int index)
...
+    case att_flavor : {
+        // if the player can't see it, forget about it
+        if (!g->u.sees(*z)) {
             break;
         }
-        if( seen ) {
-            add_msg( m_warning, flavtext, z->name().c_str() );
+
+        auto const i = get_random_index(m_flavor);
+        add_msg(m_warning, m_flavor[i], z->name().c_str());
```
Comment (narc0tiq) [inline src/monattack.cpp]: The flavor texts, on the other hand, I really like in this container format. Really easy to JSONify from there.

File: src/monattack.cpp (hunk 1 above)
Comment (KA101) [inline src/monattack.cpp]: Seconded.  Not everyone's going to see "inverse chance" and think one_in().

**Comment by KA101** [comment]
This ready to go?

**Comment by bkentel** [comment]
Yeah, I don't see why not. The helper functions can be moved in another pull request later.

**Pull request merged**